### PR TITLE
faster sv.MaskAnnotator

### PR DESCRIPTION
**Pull request by kapter**
### Description
This PR is related to https://github.com/roboflow/supervision/issues/417
I've made some enhancements to sv.MaskAnnotator.
1. calc bgr color once for color mask creation
2. replace np.where with cv2.add operation for scene with hole and colored object

### Result 
speed up two times

### CODE REVIEW

File: supervision/annotators/core.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def __init__(
         color: Union[Color, ColorPalette] = ColorPalette.default(),
         opacity: float = 0.5,
         color_map: str = "class",
+        ver: str = "v1",
```
Comment (SkalskiP) [inline supervision/annotators/core.py:98]: Please remove that extra argument.

**Review by SkalskiP — CHANGES_REQUESTED** [review]

**Comment by SkalskiP** [comment]
Hi @kapter 👋🏻, I've been busy making tests of this change. Here is what I got:

1. Consistency of obtained results

- `develop`
 
  download - 2023-10-06T180358 646

- `kapter:Make-MaskAnnotator-faster`

  download - 2023-10-06T180405 617

2. Drawing speed

Annotation speed depends on external factors - other processes running on the machine and varies between test runs. To validate the results, I performed 4 tests with the code in the current `develop` and 4  with the code from `kapter:Make-MaskAnnotator-faster`. I averaged the results.

- `develop`
  
  | baseline (only inference) | inference + box annotation | inference + mask annotation |
  |:-------------------------:|:--------------------------:|:---------------------------:|
  | 108 sec                   | 127 sec                    | 451 sec                     |
  | 103 sec                   | 108 sec                    | 465 sec                     |
  | 129 sec                   | 114 sec                    | 475 sec                     |
  | 128 sec                   | 127 sec                    | 476 sec                     |
  | **117 sec (average)**     | **119 sec (average)**      | **467 sec (average)**       |

  Annotation overhead: `467 - 117 = 350 `
  Per frame annotation overhead: `350 / 341 = 1.03`

- `kapter:Make-MaskAnnotator-faster`

  | baseline (only inference) | inference + box annotation | inference + mask annotation |
  |:-------------------------:|:--------------------------:|:---------------------------:|
  | 104 sec                   | 103 sec                    | 279 sec                     |
  | 99 sec                    | 98 sec                     | 296 sec                     |
  | 96 sec                    | 104 sec                    | 281 sec                     |
  | 102 sec                   | 104 sec                    | 266 sec                     |
  | **100 sec (average)**     | **102 sec (average)**      | **280 sec (average)**       |
  
  Annotation overhead: `280 - 100 = 180`
  Per frame annotation overhead: `180 / 341 = 0.52`

3. Summary:

- Results are consistent - no visual difference
- Updated `MaskAnnotator` is 0.5s per frame faster

**Review by SkalskiP — APPROVED** [review]

**Comment by SkalskiP** [comment]
@kapter I'm curious if we can make it even faster. 🤔 💭 `0.5` sec per frame is `2x` faster than the current implementation, but it is still `0.5` sec. It is far from real-time annotation.

Having said that, I'm merging this PR. It is a clear improvement.

Maybe we could somehow sacrifice the quality of annotation to get a speed boost.

**Comment by kapter** [comment]
I missed 
Check this please https://github.com/kapter/supervision/tree/MaskAnnotator_faster_again

**Pull request merged**